src/posts/2020-07-30-the-worst-decision-of-my-career.md

```diff
@@ -0,0 +1,271 @@
+it came as a solution to fault-tolerant distributed programming.
+
+> If all you have is a hammer, everything looks like a nail _Abraham Maslow_
+
+The designers of Erlang picked the right tool for the job, and most of us want
+to do the same. Unfortunately, there's usually more than one "tool" that would
+be "right," but to know which ones, you need to know what "job" you're solving.
+All of us **should** know this, but I keep learning about teams that fail to do
+it. There are so many companies, with a handful of "experienced" developers,
+that don't know what they are building, don't have a clear business yet, but
+their product is already built on complex architectures and technologies such
+as microservices, Kubernetes or event-sourcing. I know why we make these
+decisions, and I've made the same mistake myself.
+
+## Improving
+
+Every project we start from scratch is an opportunity to do it right. We'll
+think to ourselves: _This time I won't fall into the same traps! I have learned
+my lessons, I've studied the books, and I even met some of my gurus that wrote
+them! This time I'll follow "industry standards"!_
+
+_I am the "architect"; I will design the perfect system! Without me, none of
+this will be possible. Those mindless programmers have no idea what they are
+doing. I, and only I, am the true heir of Martin Fowler!_
+
+This was a bit dramatic, but I needed a break from all of that whining. I know
+it's easy to fall into these traps. It's even easier when your company raised
+money, and everyone is expecting you to deliver the absolute best product ever
+because they are paying you for it! This is why your starting team is so
+important; they have to withstand the pressure. They have to know that in order
+to build the grand vision, they have to go one step at a time.
+
+I was going to write a whole section on what makes for a good starting team,
+but that's a different topic, and there's still so much I need to say.
+
+## The decision
+
+I've made a lot of bad decisions, and I've been fortunate enough to suffer the
+consequences of those decisions. A lot of developers don't get to experience
+that, so they never learn. That is also why I'm always skeptical about external
+"expert" consultants; they usually don't stick around to experience the
+suffering.
+
+For many years, we, Subvisual, built web products with some success, so we felt
+confident. As if there was nothing we couldn't do. We've already learned a lot,
+so we know what to avoid. Never again, we said!
+
+So what was the worst decision of my career? It's clear to me - although some
+will disagree - that the worst decision I made was starting a product with
+event-sourcing, asynchronous communications, event-driven architecture, and all
+of that. Mind you, I didn't make the decision alone, but that doesn't change
+anything.
+
+Why was that the worst decision? Because we didn't know the market. We didn't
```

Comment:
I kind of agree with the idea that because of it being harder to change than a "regular" system, event sourced systems might not be a great solution when you are still learning the field. You want to optimize for rate of change and ES prevents it somewhat. I really don't like calling it the "worst decision ever". I think it's too much of a hyperbole. 

What about the decision of letting Oncostats linger for years with no end in sight? Or other things we built, knowing perfectly well they were a waste of money? Those are also decision you (and all of us) made not to stand your ground and stop them on their tracks. I guess you mean technical decision (even though you never say it), but even then, what about our previous website for example, when we were experimenting with SUITCSS and went completely crazy? 

I get that you have strong feelings about this one, but I don't think it is the worst decision of your career, nor do I think it should be portrayed that way. It's a click baity title that is more likely to offend people we know than to attract people we don't.

Reply:
I actually like the title a lot :p but sure, there are strong contenders to the worst decision title, but it doesn't seem like a stretch for it to be this one.

What's the matter with SUITCSS? It's still our standard for non-component systems. We used it in many places and even now we are teaching it on the summer camp.

Reply:
We went completely overboard with it, to a point that was crazy (in my opinion). If you remember we needed like 1 thousands divs to write a title, because of placing things and stuff like that.

I'm not a fan of the title a all, to be honest. It's very negative and I think the Subvisual's blog message should be a positive one, even when talking about when we mess up.

Reply:
Yes, it's only my opinion, not sure what other people think about it.